powerline-separators.el

```diff
@@ -93,11 +93,11 @@ destination color, and 2 is the interpolated color between 0 and 1."
                    `((pattern-height (max (- height ,reserve) 0))
                      (second-pattern-height (/ pattern-height 2))
                      (pattern-height ,(if second-pattern '(ceiling pattern-height 2) 'pattern-height)))
-                   `((mapconcat 'identity ',header "")
-                     (mapconcat 'identity (cl-subseq ',pattern 0 pattern-height) "")
-                     (mapconcat 'identity ',center "")
-                     (mapconcat 'identity (cl-subseq ',second-pattern 0 second-pattern-height) "")
-                     (mapconcat 'identity ',footer "")))))
+                   (list (when header `(mapconcat 'identity ',header ""))
+                         `(mapconcat 'identity (cl-subseq ',pattern 0 pattern-height) "")
```

Comment:
hrm, I see. but can pattern be nil? 

just FYI, this code was when I was being more clever than smart, so this code is kinda write once. lol. but it seems like we'll just commit this because I now see why it makes sense.